### ISSUE TITLE
COB is always zero when rewinding/scrolling back

### ISSUE THREAD

**Issue by naromero77** [nightscout/cgm-remote-monitor#5241]
**Describe the bug**
I believe this worked in the recent master but not in dev. The **current** COB pill displays correct value and updates correctly as time progress. However, if you rewind/scroll back the NS viewer you to a prior time the COB pill always displays zero. 

**To Reproduce**
Set-up is NS dev from a couple of weeks ago. Recent version of Loop dev.

Take NS viewer in a Firefox or another browser and just scroll back: COB is displaying zero, but the current instantaneous COB is correct.

**Expected behavior**
If you rewind/scroll back the NS viewer, I expect the COB displayed to show the correct value at the prior time.

**Screenshots**

**Your setup information**
- NS dev
- G6 Sharebridge
- FireFox

**Additional context**
I can add a screenshot, but it would not be interesting. It should always shows "0" in the COB pill when you scroll back. In the present time, COB is displaying correctly.

**Comment by woodduckie** [comment]
I've observed this issue after updating to v13.  I can't be certain it wasn't there before and I haven't yet had the time to roll back my deployment.  To add to the original issue, IOB reflects historical values when scrolling, but COB just shows 0.  FWIW, I'm using loop, Firefox /Safari, G6 via bridge.

**Comment by Sophieinthecloud** [comment]
![NS_COB](https://user-images.githubusercontent.com/10387075/88825015-e74d3200-d17b-11ea-8abb-68b2d4e178aa.png)

I'm having this same issue. When scrolling back in time, The COB shows in the Loop pill when I click it but the COB pill shows 0(see screenshot #1). 
A new bolus does show the COB (see screenshot #2). 

I'm not exactly sure when it started but I first noticed it after I cleaned mLab. Not sure if that is related or a coincidence.

**Comment by bjornoleh** [comment]
@sulkaharo I believe this was mentioned in CGM in the cloud quite recently. Not sure why the issue was closed just now, it still could use some attention.

https://github.com/nightscout/cgm-remote-monitor/issues/6916 mentions a possible fix for that particular issue:
_COB: buggfix to show history COB when scrolling the timeline backwards_

**Comment by naromero77** [comment]
It looks like the bug fix should go in the DIY software which is uploading information into NS. Thus the fix would have to go into OpenAPS, AndroidAPS, and Loop.

**Comment by sulkaharo** [comment]
@naromero77 Nope, this is a bug in the COB plugin not fetching the COM data from the device statuses when timeline is being scrolled.

**Comment by jgefaell** [comment]
This behavior is still present in Nightscout 14.2.6

It would be very appreciated if it could be addressed. Thank you.

**Comment by bjornoleh** [comment]
The fix mentioned in #6916 still works great

**Comment by bjornoleh** [comment]
I copied the fix mentioned previously, and created PR #8081. Credits to @klalle / @maja-lofgren

**Comment by bewest** [comment]
Thanks, @bjornoleh!

**Comment by AndyLow91** [comment]
Reviewing this old Issue and PR's associated with it.

Thanks for the reports and follow-up examples here. I’ve opened PR #8492 to address this.

The issue appears to be that Nightscout was able to find historical COB from device status data, but then rejected it because the freshness check compared the COB timestamp against the real current time instead of the time being viewed on the chart.

That meant when scrolling back, Nightscout could fall back to calculating COB from treatments instead of using the COB reported by the controller. In practice that can show `0g`, or a nearby but different value from what appears in the OpenAPS/Loop devicestatus data.

The PR changes the COB plugin so that historical COB is checked relative to the timeline position being viewed. It uses the existing device-status COB parsing, so this covers both OpenAPS/iAPS/AAPS-style COB and Loop-style COB rather than adding a special case for only one uploader format.

I also added regression tests for both OpenAPS-style and Loop-style historical COB, plus a fallback test to make sure stale historical data still falls back safely to treatment-derived COB.
